presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -805,18 +805,19 @@ public void testShowCatalogs()
     @Test
     public void testShowSchemas()
     {
-        assertStatement("SHOW SCHEMAS", new ShowSchemas(Optional.empty(), Optional.empty()));
-        assertStatement("SHOW SCHEMAS FROM foo", new ShowSchemas(Optional.of(identifier("foo")), Optional.empty()));
-        assertStatement("SHOW SCHEMAS IN foo LIKE '%'", new ShowSchemas(Optional.of(identifier("foo")), Optional.of("%")));
+        assertStatement("SHOW SCHEMAS", new ShowSchemas(Optional.empty(), Optional.empty(), Optional.empty()));
+        assertStatement("SHOW SCHEMAS FROM foo", new ShowSchemas(Optional.of(identifier("foo")), Optional.empty(), Optional.empty()));
+        assertStatement("SHOW SCHEMAS IN foo LIKE '%'", new ShowSchemas(Optional.of(identifier("foo")), Optional.of("%"), Optional.empty()));
+        assertStatement("SHOW SCHEMAS IN foo LIKE '%$_%' ESCAPE '$'", new ShowSchemas(Optional.of(identifier("foo")), Optional.of("%$_%"), Optional.of("$")));
```

Comment:
Add a test with escape being empty and also one with escape longer than 1 character. (Should fail.. if not at parsing stage then at some later stage.)

Also add test cases `com.facebook.presto.tests.AbstractTestQueries#testShowSchemasLike` and `com.facebook.presto.tests.AbstractTestQueries#testShowTablesLike`.. or better yet to new methods `AbstractTestDistributedQueries.{testShowSchemasLikeWithEscape,testShowTablesLikeWithEscape}`, there you would be able to create a table with `_`, just as in `com.facebook.presto.tests.AbstractTestDistributedQueries#testCreateTable`

Reply:
@findepi Thank you for your good ideas. I add all tests in com.facebook.presto.tests.AbstractTestDistributedQueries.{testShowSchemasLikeWithEscape,testShowTablesLikeWithEscape}